### PR TITLE
Update RARBG to new API (v2)

### PR DESCRIPTION
**Pull request by fernandog**

### CODE REVIEW

File: sickbeard/providers/rarbg.py [hunk 1]
```diff
@@ -65,11 +65,11 @@ def __init__(self):
...
-                     'listing': u'https://torrentapi.org/pubapi.php?mode=list',
-                     'search': u'https://torrentapi.org/pubapi.php?mode=search&search_string={search_string}',
-                     'search_tvdb': u'https://torrentapi.org/pubapi.php?mode=search&search_tvdb={tvdb}&search_string={search_string}',
-                     'search_tvrage': u'https://torrentapi.org/pubapi.php?mode=search&search_tvrage={tvrage}&search_string={search_string}',
+                     'token': u'https://torrentapi.org/pubapi.php?get_token=get_token&format=json&app_id=sickrage',
+                     'listing': u'https://torrentapi.org/pubapi.php?mode=list&app_id=sickrage',
+                     'search': u'https://torrentapi.org/pubapi.php?mode=search&app_id=sickrage&search_string={search_string}',
+                     'search_tvdb': u'https://torrentapi.org/pubapi.php?mode=search&app_id=sickrage&search_tvdb={tvdb}&search_string={search_string}',
+                     'search_tvrage': u'https://torrentapi.org/pubapi.php?mode=search&app_id=sickrage&search_tvrage={tvrage}&search_string={search_string}',
                      'api_spec': u'https://rarbg.com/pubapi/apidocs.txt',
```
Comment (fernandog) [inline sickbeard/providers/rarbg.py]: @rarbg like this?

**Comment by miigotu** [comment]
@fernandog AFAIK we need to use `pubapi_v2.php`

**Comment by fernandog** [comment]
@miigotu done

**Comment by miigotu** [comment]
Cool, I'll test it after the buildtest and it is merged

**Comment by miigotu** [comment]
@fernandog the field names in the json have changed too in v2, so the parsing of results also has to be changed.

**Comment by miigotu** [comment]
Example result v2:
{"torrent_results":[{"filename":"Discovery.Channel.Aftershock.Disaster.In.Nepal.720p.HDTV.x264-CBFM[rartv]","category":"TV HD Episodes","download":"magnet:?xt=urn:btih:9aa4fc4aa17ebaef2e5788780269ffe0152a0950&dn=Discovery.Channel.Aftershock.Disaster.In.Nepal.720p.HDTV.x264-CBFM%5Brartv%5D&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2710&tr=udp%3A%2F%2F9.rarbg.to%3A2710&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce"}]}

**Comment by miigotu** [comment]
There are more fields too if you use json_extended, or other search options.

**Comment by fernandog** [comment]
@miigotu done

do we need the others?

**Comment by miigotu** [comment]
Not for now @fernandog it just needs to work as it did before the api change. Soonish, I plan on improving the code there with new improvements, that will use the json_extended attributes, and also reduce server load. As it sits now we hammer rarbg because a separate request is made for each episode we need for each show.

This is a worst case scenario guesstimation:
100 Shows * 10 Needed Eps * 1000 Users every 15 minutes??  1,000,000 requests / 15 mins, or 1100 requests a second * 12KB = 13MB/s

**Comment by miigotu** [comment]
Granted we have rate limiting, both on the api and in SR, but you get what I mean.

**Comment by fernandog** [comment]
@miigotu why multiply by 100 Shows? don't need. only x10 (needed episodes)

**Comment by miigotu** [comment]
What if you have 100 shows and you need 10 episodes for each one?

**Comment by miigotu** [comment]
It's common practice to over-estimate bandwidth and processing costs in the computing world.

**Comment by fernandog** [comment]
@miigotu squashed

**Comment by miigotu** [comment]
rarbg/torrentapi#1

**Comment by rarbg** [comment]
@miigotu @fernandog 
Exactly 2043 unique ips hit the api with *sickrage* in the user agent the last 24h

**Comment by miigotu** [comment]
@rarbg Do your statistics show requests/ip?

**Comment by rarbg** [comment]
@miigotu 
1. The stats are not ready yet
2. Those 2043 are from v1+v2 , not only v2 :)
3. Please use correct user-agent strings because of cloudflare ( aka sickrage/whatever )

**Pull request merged**